Defining panic function with incorrect signature in tests crashes compiler

```rust
// This test have to be here, or we get a correct error message:
//   test3.zig:5:5: error: expected 'fn([]const u8, ?*builtin.StackTrace) noreturn', found 'fn() void'
//   pub fn panic() void {}
test "" {}

pub fn panic() void {}
```
```
zig/src/analyze.cpp:465: ZigType* get_pointer_to_type_extra(CodeGen*, ZigType*, bool, bool, PtrLen, uint32_t, uint32_t, uint32_t): Assertion `type_is_resolved(child_type, ResolveStatusZeroBitsKnown)' failed.
[1]    15395 abort (core dumped)  zig1 test test3.zig
```

--- a/src/analyze.cpp
+++ b/src/analyze.cpp
@@ -1058,7 +1058,10 @@ ZigType *get_ptr_to_stack_trace_type(CodeGen *g) {
     if (g->stack_trace_type == nullptr) {
         ConstExprValue *stack_trace_type_val = get_builtin_value(g, "StackTrace");
         assert(stack_trace_type_val->type->id == ZigTypeIdMetaType);
+
         g->stack_trace_type = stack_trace_type_val->data.x_type;
+        assertNoError(ensure_complete_type(g, g->stack_trace_type));
+
         g->ptr_to_stack_trace_type = get_pointer_to_type(g, g->stack_trace_type, false);
     }
     return g->ptr_to_stack_trace_type;
@@ -6560,4 +6563,3 @@ uint32_t get_host_int_bytes(CodeGen *g, ZigType *struct_type, TypeStructField *f
     LLVMTypeRef field_type = LLVMStructGetTypeAtIndex(struct_type->type_ref, field->gen_index);
     return LLVMStoreSizeOfType(g->target_data_ref, field_type);
 }
-

--- a/src/analyze.cpp
+++ b/src/analyze.cpp
@@ -1060,7 +1060,7 @@ ZigType *get_ptr_to_stack_trace_type(CodeGen *g) {
         assert(stack_trace_type_val->type->id == ZigTypeIdMetaType);
 
         g->stack_trace_type = stack_trace_type_val->data.x_type;
-        assertNoError(ensure_complete_type(g, g->stack_trace_type));
+        assertNoError(type_resolve(g, g->stack_trace_type, ResolveStatusZeroBitsKnown));
 
         g->ptr_to_stack_trace_type = get_pointer_to_type(g, g->stack_trace_type, false);
     }

--- a/test/compile_errors.zig
+++ b/test/compile_errors.zig
@@ -2,6 +2,16 @@ const tests = @import("tests.zig");
 
 pub fn addCases(cases: *tests.CompileErrorContext) void {
     cases.add(
+        "Panic declared with wrong type signature in tests",
+        \\test "" {}
+        \\
+        \\pub fn panic() void {}
+        \\
+    ,
+        ".tmp_source.zig:3:5: error: expected 'fn([]const u8, ?*builtin.StackTrace) noreturn', found 'fn() void'",
+    );
+
+    cases.add(
         "compile error when evaluating return type of inferred error set",
         \\const Car = struct {
         \\    foo: *SymbolThatDoesNotExist,

